ocaml/xapi/repository.ml

```diff
@@ -511,50 +511,51 @@ let get_host_updates_in_json ~__context ~installed =
         (ExnHelper.string_of_exn e) ;
       raise Api_errors.(Server_error (get_host_updates_failed, [ref]))
 
+let response_repo_file ~req ~s uri_path =
+  if Fileserver.access_forbidden req s then
```

Comment:
access privilege should be checked before resolve path? (`map_uri_to_path` is called before `response_repo_file` now)